Cannot install happy because happy is required

`happy` is in a circular dependency with itself, see below.

What I did is:

```
sudo apt-get install all-the-required-stuff
git clone https://github.com/GaloisInc/HaLVM.git
git submodule update --init --recursive
autoconf
./configure --enable-gmp
make
```

`make` step tries to install `happy` and it fails.

```
vagrant@vagrant-ubuntu:/vagrant/HaLVM$ make
PATH=/vagrant/HaLVM/platform_ghc/bin:/vagrant/HaLVM/build/sandbox/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games && cd /vagrant/HaLVM/build/happy-1.19.5 && /vagrant/HaLVM/platform_ghc/usr/local/lib/HaLVM-2.0.0/bin/cabal --config-file="/vagrant/HaLVM/build/cabal.config" sandbox init --sandbox /vagrant/HaLVM/build/sandbox && /vagrant/HaLVM/platform_ghc/usr/local/lib/HaLVM-2.0.0/bin/cabal --config-file="/vagrant/HaLVM/build/cabal.config" install
Writing a default package environment file to
/vagrant/HaLVM/build/happy-1.19.5/cabal.sandbox.config
Using an existing sandbox located at /vagrant/HaLVM/build/sandbox
Resolving dependencies...
Configuring happy-1.19.5...
Building happy-1.19.5...
Preprocessing executable 'happy' for happy-1.19.5...
setup: The program 'happy' is required but it could not be found
Failed to install happy-1.19.5
cabal: Error: some packages failed to install:
happy-1.19.5 failed during the building phase. The exception was:
ExitFailure 1
make: *** [/vagrant/HaLVM/build/sandbox/bin/happy] Error 1
```

--- a/Makefile
+++ b/Makefile
@@ -122,7 +122,10 @@ $1_BIN := $$(BUILDBOX)/bin/$2
 $$($1_BIN): $$(PLATCABAL) | $$($1_SRC)
 	$$(BUILDENV) && cd $$($1_SRC) && \
 		$$(CABAL) sandbox init --sandbox $$(BUILDBOX) && \
-		$$(CABAL) install
+                $$(CABAL) install --only-dep && \
+                $$(CABAL) configure && \
+                $$(CABAL) build && \
+                $$(CABAL) copy
 endef
 $(eval $(call hackage-sandbox-build,ALEX,alex))
 $(eval $(call hackage-sandbox-build,HAPPY,happy))

--- a/README.md
+++ b/README.md
@@ -56,10 +56,6 @@ You'll need these dependencies:
 
 > autoconf gcc automake libtool patch ncurses-devel xen-devel zlib-devel
 
-Additionally, make sure `alex >= 3.1` and `happy >= 1.19` are installed and
-in your PATH. This is a workaround for a 
-[Cabal bug](https://github.com/haskell/cabal/issues/2462).
-
 If GHC 7.8.4 is not present, a version of it will be downloaded for
 use during the build process.
 

